hphp/hack/src/hackfmt/hack_format.ml

```diff
@@ -822,7 +822,15 @@ let rec t (env: Env.t) (node: Syntax.t) : Doc.t =
       t env kw;
       Space;
       transform_condition env left_p condition right_p;
-      handle_possible_compound_statement env if_body;
+      (match Syntax.syntax if_body with
+      | Syntax.CompoundStatement _ ->
+          handle_possible_compound_statement env if_body;
+      | _ -> Concat [
+          Space;
+          SplitWith Cost.Base;
+          Nest [t env if_body];
```

Comment:
@alisha, I'm seeing some pretty funky-looking results from some of our internal test cases.

Code that was previously formatted like this:

```
if ($this->head->isEmpty())
  return new MyQueue($this->head->push($item), $this->tail);
else
  return new MyQueue($this->head, $this->tail->push($item));
```

Is now formatted like this:

```
if ($this->head->isEmpty()) return
  new MyQueue($this->head->push($item), $this->tail);
else return new MyQueue($this->head, $this->tail->push($item));
```

We could probably deal with this case:

```
if ($cond) return
  $x;
```

by using a `Parental` rule to control the split before the body instead of an independent split (`SplitWith`):

```
WithRule (Rule.Parental, Nest [ Span [
  Space;
  Split;
  t env if_body;
]]);
```

Using a `Parental` rule means that if any line break is added within the body, we'll add a line break at the `Split` preceding the body, too. Sorry for suggesting `SplitWith` earlier--I hadn't considered this possible outcome!

Even if we deal with that case, to be frank, I think this formatting:

```
if ($this->head->isEmpty())
  return new MyQueue($this->head->push($item), $this->tail);
else
  return new MyQueue($this->head, $this->tail->push($item));
```

Is much more readable than this:

```
if ($this->head->isEmpty())
  return new MyQueue($this->head->push($item), $this->tail);
else return new MyQueue($this->head, $this->tail->push($item));
```

Or this:

```
if ($empty) return new MyQueue($this->head->push($item), $this->tail);
else return new MyQueue($this->head, $this->tail->push($item));
```

So I've become a little more uneasy about this change! I feel like some human input is useful in deciding when it's appropriate to put a single-statement `if` on one line.

One way we could deal with this is by preserving single-line `if`s when they already exist (and fit within the line length limit), and not forcing other single-statement `if`s onto one line if they aren't already. We just need to find out if there was already a newline after the condition's right parenthesis:

```
if has_newline (Syntax.trailing_trivia right_p)
then Newline
else Split;
```

Since this is starting to be a fair bit of code which is duplicated across `IfStatement`, `ElseifClause`, and `ElseClause`, perhaps we should move it to a new helper function called something like `transform_consequence` :)